packages/web-react/src/components/Stack/useStackStyleProps.ts

```diff
@@ -0,0 +1,32 @@
+import classNames from 'classnames';
+import { ElementType } from 'react';
+import { useClassNamePrefix } from '../../hooks';
+import { SpiritStackProps } from '../../types';
+
+export interface StackStyles {
+  /** className props */
+  classProps: string;
+  /** props to be passed to the element */
+  props: unknown;
+}
+
+export function useStackStyleProps<T extends ElementType = 'div'>(props: SpiritStackProps<T>): StackStyles {
+  const { hasBottomDivider, hasMiddleDividers, hasSpacing, hasTopDivider, ...restProps } = props;
+
+  const StackClass = useClassNamePrefix('Stack');
+  const StackBottomDividerClass = `${StackClass}--hasBottomDivider`;
+  const StackMiddleDividersClass = `${StackClass}--hasMiddleDividers`;
```

Comment:
There is an inconsistency in class naming, I think. `Divider` vs `Dividers` for the middle. I think it is intentional but can be a source of mistakes.

Reply:
You're right, but I think that the name should be descriptive as much as possible, and the name `hasMiddleDivider` wouldn't match to the result.